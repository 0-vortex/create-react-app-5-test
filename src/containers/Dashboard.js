import React, {Suspense} from "react";
import Repository from "../components/Repository";
import RepositoryGoals from "../components/RepositoryGoals";
import {Route} from "react-router-dom";
import {Spinner} from "../styles/Spinner";

function Repositories({user, match}) {
  return (
    <Suspense fallback={<Spinner />}>
      <Route exact path={match.url} render={(props) => <RepositoryGoals user={user} {...props} />} />
      <Route path={"/repos/:repoOwner/:repoName/:id"} render={(props) => <Repository user={user} {...props} />} />
    </Suspense>
  );
}

export default Repositories;
