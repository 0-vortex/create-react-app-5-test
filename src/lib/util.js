export function isValidRepoUrl(url) {
  url = url.trim();
  url = url.substr(0, 1) === "/" ? url.substr(1) : url;

  const isRelativeUrl = !(url.substr(0, 4) === "http" || url.includes(".com") || url.includes("www."));
  if (isRelativeUrl) {
    return relativeUrlValidator(url);
  }

  return absoluteUrlValidator(url);

}

function relativeUrlValidator(url) {
  try {
    const githubLink = "github.com/";
    const newUrl = "https://" + githubLink + url;
    const urlObject = new URL(newUrl);

    const [owner, repo] = url.split("/");
    if (!owner || !repo || !(urlObject.protocol === "http:" || urlObject.protocol === "https:")) {
      return [false, null];
    }
    return [true, url];
  } catch (failedToConstructURL) {
    return [false, null];
  }
}

function absoluteUrlValidator(url) {
  try {
    const githubLink = "github.com/";
    const urlObject = new URL(url);

    const getIndexGithub = url.indexOf(githubLink) + githubLink.length;
    const relativeRepoUrl = url.substr(getIndexGithub, url.length);

    const [owner, repo] = relativeRepoUrl.split("/");

    if (urlObject.hostname !== "github.com") {
      return [false, null];
    }
    if (!owner || !repo || !(urlObject.protocol === "http:" || urlObject.protocol === "https:")) {
      return [false, null];
    }
    return [true, owner + "/" + repo];
  } catch (failedToConstructURL) {
    return [false, null];
  }
}
