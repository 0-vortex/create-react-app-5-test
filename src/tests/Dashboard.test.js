import React from "react";
import {shallow} from "enzyme";
import Dashboard from "../containers/Dashboard";

import {LocalStorageMock, data, match} from "./mocks";
import {promisify} from "util";

// set up local storage mock usage in component
global.localStorage = new LocalStorageMock();
localStorage.setItem("currentOpenSaucedUser", JSON.stringify({...data}));

const emptyPromise = promisify(() => {});

describe("<Dashboard />", () => {
  it("should render without throwing an error", () => {
    const component = shallow(
      <Dashboard createViewer={emptyPromise} match={match} data={data} />,
    );
    expect(component).toMatchSnapshot();
  });
});
