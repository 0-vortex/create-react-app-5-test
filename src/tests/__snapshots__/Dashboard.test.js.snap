// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> should render without throwing an error 1`] = `ShallowWrapper {}`;
